report/3.Theory/LanguageSpecification.tex

```diff
@@ -34,26 +28,44 @@ \subsection{Types}
 Types can be declared in a few different ways, as the preliminary interviews indicate that users might be interested in declaring types both explicitly and implicitly.
 Therefore, types can be declared using the keyword \textbf{\textit{var}} (short for \textbf{variable\textit{}}) prefixed to an assignment, listing \ref{lis:typedcls}, line 1.
 Declarations can also be explicit by prefixing the type of the variable before an assignment, listing \ref{lis:typedcls}, line 2, or by prefixing the type before the variable without an assignment, listing \ref{lis:typedcls}, line 3.
-The three methods of declaration are seen in snippet below.
-A  \textbf{\textit{num}} example is also shown.
+
+Declaration of the different types are done in the following way:
+\begin{lstlisting}[language=HCL,label=lis:typedcls,firstnumber=1]
+num x = 10
+txt y = "string"
+bool z = True
+list[num] w = [0,1,1.2]
+tuple[txt,num] v = ("hello",2)
+func[txt] u = ():txt { "hello" }
+\end{lstlisting}
+
+As previously stated, declaration can be done either with an implicit type and an assignment, explicit type and an assignment or explicit type.
+This is shown through the \textbf{\textit{num}} type, in the following example:
 
 \begin{lstlisting}[language=HCL,label=lis:typedcls,firstnumber=1]
 var x = 5
 num y = 10
-txt s = "string"
-bool b
+num z
+
 \end{lstlisting}
 
 Using the prefix \textbf{\textit{var}} enables the use of type-inference, where the type of the variable is automatically inferred at compile time \cite{typeinf}.
 HCL uses static typing, meaning that type-checking is done at compile time.
 
 \subsection{Operators}
-Most of the operators in HCL are defined as functions.
+Most of the typical operators found in structural and object oriented languages are defined as built-in functions in HCL.
```

Comment:
1)
Det er all operators
2)
 kendte fra object orienteret languages?
de vel kendt fra... matematik... og generelle programmerings sprog :P "tructural and object oriented" virker sygt misvisenede

Reply:
True "Fra matematik" er i hvert fald bedre. Fixed